lib/child_process.js

```diff
@@ -754,14 +768,25 @@ function sanitizeKillSignal(killSignal) {
 // This level of indirection is here because the other child_process methods
 // call spawn internally but should use different cancellation logic.
 function spawnWithSignal(file, args, options) {
-  const child = spawn(file, args, options);
+  // Remove signal from options to spawn
+  // to avoid double emitting of AbortError
+  let child;
+
+  if (options && typeof options === 'object' && ('signal' in options)) {
+    const opts = { ...options };
+    delete opts.signal;
+
+    child = spawn(file, args, opts);
+  } else {
+    child = spawn(file, args, options);
+  }
```

Comment:
I think this achieve the same result without performing a `delete` operation.
```suggestion
  const child = spawn(file, args, { ...options, signal: undefined });
```

Reply:
Fun fact, that was my first approach but wasn't sure if it was performant 😅 

Even tho, we need to check that the options are an `object` and contain a `signal` property inside. If `null` or `number` is passed, the distructing will pass and the `spawn` function will receive the object `{ signal: undefined }` which shouldn't happen as by design we should forward the `number` or `null` and let `spawn` throw when validating the options.

If I move with your suggestion some tests will fail because of the logic I described, what do you think?

Reply:
> If `null` or `number` is passed, the distructing will pass and the `spawn` function will receive the object `{ signal: undefined }` which shouldn't happen as by design we should forward the `number` or `null` and let `spawn` throw when validating the options.

I don't really get this, in the current code, you delete the `signal` property without checking its type, how is my suggestion different on that point?

> we need to check that the options are an `object`

OK, that seems like something we need to check indeed.

What do you think of that?

```suggestion
  const child = spawn(file, args, typeof options === 'object' && options !== null && 'signal' in options ?
    { ...options, signal: undefined } :
    options);
```

Reply:
Sure, has sense 👍

Reply:
Done dade71b
Please have a look